lstchain/tools/tests/test_tools.py

```diff
@@ -212,6 +242,38 @@ def test_create_dl3_with_config(temp_dir_observed_files, observed_dl2_file):
     )
 
 
+@pytest.mark.private_data
+def test_create_srcdep_dl3(temp_dir_observed_srcdep_files, observed_srcdep_dl2_file, simulated_srcdep_irf_file):
+    """
+    Generating a source-dependent DL3 file from a test DL2 files and test IRF file
+    """
+    from lstchain.tools.lstchain_create_dl3_file import DataReductionFITSWriter
+
+    assert (
+        run_tool(
+            DataReductionFITSWriter(),
+            argv=[
+                f"--input-dl2={observed_srcdep_dl2_file}",
+                f"--output-dl3-path={temp_dir_observed_srcdep_files}",
+                f"--input-irf={simulated_srcdep_irf_file}",
+                "--source-name=Crab",
+                "--source-ra=83.633deg",
+                "--source-dec=22.01deg",
+                "--source-dep",
+                "--overwrite",
+            ],
+            cwd=temp_dir_observed_srcdep_files,
+        )
+        == 0
+    )
+
+    hdulist = fits.open(temp_dir_observed_srcdep_files / "dl3_LST-1.Run02008.0000.fits.gz")
```

Comment:
Use from `lstchain.paths import dl2_to_dl3_filename` to get the name of the DL3 file, for more uniform code